modules/ximgproc/include/opencv2/ximgproc/fast_line_detector.hpp

```diff
@@ -40,8 +40,10 @@ class CV_EXPORTS_W FastLineDetector : public Algorithm
       and ending point of a line.  Where Vec4f is (x1, y1, x2, y2), point
       1 is the start, point 2 - end. Returned lines are directed so that the
       brighter side is on their left.
+      @param is_edge If true, image will be considerd as edge and negrect the canny parameters.
       */
-    CV_WRAP virtual void detect(InputArray _image, OutputArray _lines) = 0;
+    CV_WRAP virtual void detect(InputArray _image, OutputArray _lines,
+            bool is_edge = false) = 0;
```

Comment:
`detect()` is not a right place for algorithm parameters tuning. See `createFastLineDetector()`

Can we check some canny parameters to handle that without touching API?
For example, `canny_aperture_size == 0`